### ISSUE TITLE
Delete/prevent generation of MyTemplate.cs if MyTemplate.tt succeeded w/o error

### ISSUE THREAD

**Issue by Eagle3386** [databinding-gmbh/T4.FileManager.VisualStudio#2]
Creating C# code files via T4.FileManager also creates a file named just like its T4 template file, but with `cs` as its extension.
Example: `MyTemplate.tt` (uses T4.FileManager) creates a `MyTemplate.cs" file, located just next to all generated C# code files.

While this is definitely useful for debug purposes, this results in 2 issues:
a) If T4 template succeeded, the file remains empty, but (at least when using Git) creates a pending change (adding the file) - using `.gitignore` as a workaround is kinda senseless as the file should've been deleted in the first place upon successful completion.
b) In case of an error, I've never seen anything useful - only `ErrorGeneratingOutput` is written to the file. If the `debug` attribute is set to `true`, T4.FileManager should print as much debug output as available to the file.

**Comment by databinding-bot-zz** [comment]
Can you provide an example of the problem (e.g. as a repository on GitHub)?

**Comment by Eagle3386** [comment]
Sorry for the delay. Yes, I can. But bear with me if it might take a day or two more as expected.

**Comment by Eagle3386** [comment]
Sorry for the delay, which I'd like to apologize for by providing anyone still interested with my findings below.

**tl;dr:**
- Change your `.tt` file's [`output` directive](https://docs.microsoft.com/en-us/visualstudio/modeling/t4-output-directive) `extension` attribute value to: `<#@ output encoding="utf-8" extension="/" #>`
Yes, it's an illegal file extension, but since both, IntelliSense & `ITextTemplatingEngineHost`, don't complain & instead silently continue processing, why bother? 😉
- Until release of a version with a configurable property, go to `line 80` of `T4.FileManager.VisualStudio.ttinclude` file & add: `this._host.SetFileExtension("/");`

... and you're done: no (empty) unwanted files anymore, no compile errors due to `ErrorGeneratingOutput` within `MyTemplate.cs` anymore, just 🥳!

**Long(er) story:**
1. The infamous "MyTemplate.cs" is created by Visual Studio's `ITextTemplatingEngineHost` & therefore _**seemed**_ impossible to fix as originally confirmed [here](https://stackoverflow.com/questions/2601202/how-not-to-output-default-t4-generated-file/2611334#2611334).
2. The ASP.NET MVC team worked around this at least since MVC 3 by implementing a custom `ITextTemplatingEngineHost` for their scaffolding as confirmed [here](https://stackoverflow.com/questions/13838142/put-all-methods-in-one-tt-file-and-use-it-in-another-t4-files-in-codetemplates/13847781#13847781).
3. However, thanks to @Athari's awesome findings over [here](https://stackoverflow.com/questions/2601202/how-not-to-output-default-t4-generated-file/13878689#13878689), I **did** managed to fix this in the end.
4. I will create a PR for extension of `T4.FileManager.VisualStudio` to enable others to utilize it, too, right after this comment.

**Comment by Eagle3386** [comment]
@db-ty Done, please see #3 & let me know what you think &/or how I can improve it to get it merged.

**Comment by Eagle3386** [comment]
@db-ty Done, please see #5 & let me know what you think &/or how I can improve it to get it merged.

**Comment by renegadexx** [comment]
Hi Martin

We added your suggestion as method DisableTemplateMainOutputFile to prevent the generation of the main output file. Call this method after the Create(this). In the constructor we do hard set the file extension to .txt, to avoid error generating output messages in the ide, if the new method is not used.

We used the separate method because the approach you suggested produce other warnings, that we do not want to have in our projects.

We refresh the nuget package in the next few days.

Thanks,
René & Johnny

**Comment by Eagle3386** [comment]
@renegadexx Thanks for the follow-up. The workaround you've added still produces the `.info.json` file which seems to modify the `.csproj` which in turn gets caught by version control systems out there - that should be improved, too.

Speaking of warning, the changes you've made don't fix the warnings I wrote about in yesterday's reply to #5 - are you planning on fixing them, too? Because otherwise, I'd still be forced to use my "PR version" of your T4 generator...

**Comment by renegadexx** [comment]
Hi Martin,

The issue describes a problem with the main output file with output extension .cs. This is solved with the assigned commit. The feature toggle which disables the t4 default output file with an invalid extension is solved too. The feature file "UT001 Generate Code" has three new scenarios which test these requirements.
 
The two warnings are solved with a downgrade of the template code to net 4 syntax. The newer syntax led to the errors in the error list of Visual Studio when you opened the template code. 
 
As described in the readme and as comment to your pull request are the info.json files required for the core functionality of the file manager. We do not want introduce feature toggles that disable this core functionality. We switched to this approach, because the old approach with .txt4 does not work anymore since Microsoft has made changes in the solution structure for .net core projects.
 
We created an issue that the info.json files make changes to the csproj. Please provide a t4 sample with details of Visual Studio, NuGet-Package and .NET Framework version, steps, a print screen, so that we can try to reproduce this issue and add it to #6.
 
Thanks

Johnny and René
